### ISSUE TITLE
Build not failing when it should

### ISSUE THREAD

**Issue by jamestalmage** [firebase/angularfire#542]
See [this build](https://travis-ci.org/firebase/angularfire/builds/48444726) for #541. There is a test error but Travis reports it as passing.

**Comment by jwngr** [comment]
Huh, I don't even understand why this happened, but I should be able to investigate later today.

**Comment by jamestalmage** [comment]
I looked into it and submitted a fix in #545. 
I think whoever wrote the bash script for the travis tests assumed the script would exit immediately on errors. That's not the case, instead it just returns and stores it in the `$?` variable. You can then use conditional logic to handle the returned error. Traditionally you'd do this:

```bash
grunt test:unit
if [ "$?" != "0" ]; then
  echo "error in unit tests!"
  exit 1
fi
# or alternate shorthand
grunt test:unit || exit 1
```

That gets old really fast, so there's now a `set -e` flag which makes your script throw abort with an error code any time it encounters an error. It [has some gotchas](http://mywiki.wooledge.org/BashFAQ/105), but I think it's save to use in this case.
